src/index.js

```diff
@@ -19,15 +19,21 @@ const AsyncComputed = {
       .asyncComputed = Vue.config.optionMergeStrategies.computed
 
     Vue.mixin({
+      data () {
+        return {
+          _asyncComputed: {},
+        }
+      },
       beforeCreate () {
         const optionData = this.$options.data
         const asyncComputed = this.$options.asyncComputed || {}
-        this.$asyncComputed = {}
-
-        if (!Object.keys(asyncComputed).length) return
 
         if (!this.$options.computed) this.$options.computed = {}
 
+        this.$options.computed.$asyncComputed = () => this.$data._asyncComputed
```

Comment:
Note that I had to implement a little hack here using `_asyncComputed` in `$data` and a corresponding `$asyncComputed` computed property to maintain backwards compatibility. This is necessary since Vue does not proxy properties that start with `_` or `$`.

Alternatives, if backwards compatibility is not needed are:
1. Rename `$asyncComputed` as simply `asyncComputed`.
2. Keep only `$asyncComputed` in data instead of `_asyncComputed`, drop the unnecessary computed property and document that it should be accessed as `this.$data.$asyncComputed`.